src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -660,49 +667,45 @@ private LocalSymbol GetDeclaredLocal(CSharpSyntaxNode declarationSyntax, SyntaxT
                 {
                     if (local.IdentifierToken == declaredIdentifier)
                     {
-                        return GetAdjustedLocalSymbol(local, declarationSyntax.SpanStart);
+                        return GetAdjustedLocalSymbol((SourceLocalSymbol)local);
                     }
                 }
             }
 
             return null;
         }
 
-        internal override LocalSymbol GetAdjustedLocalSymbol(LocalSymbol local, int position)
+#nullable enable
+        internal override LocalSymbol GetAdjustedLocalSymbol(SourceLocalSymbol local)
         {
-            Debug.Assert(local is SourceLocalSymbol);
-            LocalSymbol adjustedLocal;
-            if (Compilation.NullableSemanticAnalysisEnabled)
-            {
-                if (!_analyzedVariableTypesOpt.TryGetValue(local, out adjustedLocal))
-                {
-                    var types = GetSnapshotManager().GetVariableTypesForPosition(position);
-
-                    // If the local was not inferred, it does not get an entry in this dictionary. Save the local mapped
-                    // to itself to avoid needing to enter this code path in the future.
-                    if (types.TryGetValue(local, out TypeWithAnnotations type))
-                    {
-                        adjustedLocal = _analyzedVariableTypesOpt.GetOrAdd(local, ((SourceLocalSymbol)local).WithAnalyzedType(type));
-                    }
-                    else
-                    {
-                        _analyzedVariableTypesOpt.TryAdd(local, local);
-                        adjustedLocal = local;
-                    }
-                }
-            }
-            else
-            {
-                adjustedLocal = local;
-            }
+            return GetRemappedSymbol<LocalSymbol>(local);
+        }
 
-            return adjustedLocal;
+        private LocalFunctionSymbol GetDeclaredLocalFunction(LocalFunctionStatementSyntax declarationSyntax)
+        {
+            // If nullable semantic analysis is enabled, we need to potentially reinfer the containing symbol
+            // of a local function, as it could be in a reinferred lambda. For an example, see
```

Comment:
The comment makes sense but it doesn't look like we're explicitly reinferring the containing symbol here. (That just happens to be a necessary part of reinferring the local function.) Consider removing the comment.